slideshow/slideshow.go

```diff
@@ -0,0 +1,155 @@
+		}
+		var motions = [][]float64{}
+		if slide.Motion.Start == "" {
+			motions = [][]float64{{0, 0, 1, 1}, {0, 0, 1, 1}}
+		} else {
+			motions = [][]float64{helper.ConvertStringToFloat(slide.Motion.Start), helper.ConvertStringToFloat(slide.Motion.End)}
+		}
+
+		Motions = append(Motions, motions)
+		Timings = append(Timings, slide.Timing.Duration)
+	}
+
+	slideshow := slideshow{Images, Audios, Transitions, TransitionDurations, Timings, Motions, template_name}
+
+	fmt.Println("Parsing completed...")
+
+	return slideshow
+}
+
+/* Function to scale all the input images depending on video quality
+ * option to a uniform height/width to prevent issues in the video creation process.
+ */
+
+func (s slideshow) ScaleImages(lowQuality *bool) {
+	width := "1280"
+	height := "720"
+
+	if *lowQuality {
+		width = "852"
+		height = "480"
+	}
+
+	totalNumImages := len(s.images)
+	var wg sync.WaitGroup
+	// Tell the 'wg' WaitGroup how many threads/goroutines
+	//   that are about to run concurrently.
+	wg.Add(totalNumImages)
+
+	for i := 0; i < totalNumImages; i++ {
+		go func(i int) {
+			defer wg.Done()
+			cmd := FFmpeg.CmdScaleImage(s.images[i], height, width, s.images[i])
+			output, err := cmd.CombinedOutput()
+			FFmpeg.CheckCMDError(output, err)
+		}(i)
+	}
+
+	wg.Wait()
+}
+
+func (s slideshow) CreateVideo(useOldfade *bool, tempDirectory string, outputDirectory string) {
+	// Checking FFmpeg version to use Xfade
+	fmt.Println("Checking FFmpeg version...")
+	var fadeType string = FFmpeg.CheckVersion()
+	useXfade := fadeType == "X" && !*useOldfade
+
+	final_template_name := strings.TrimSuffix(s.templateName, ".slideshow")
+
+	if useXfade {
+		fmt.Println("FFmpeg version is bigger than 4.3.0, using Xfade transition method...")
+		FFmpeg.MakeTempVideosWithoutAudio(s.images, s.transitions, s.transitionDurations, s.timings, s.audios, s.motions, tempDirectory)
+		FFmpeg.MergeTempVideos(s.images, s.transitions, s.transitionDurations, s.timings, tempDirectory)
+		FFmpeg.AddAudio(s.timings, s.audios, tempDirectory)
+		FFmpeg.CopyFinal(tempDirectory, outputDirectory, final_template_name)
+	} else {
+		fmt.Println("FFmpeg version is smaller than 4.3.0, using old fade transition method...")
+		FFmpeg.MakeTempVideosWithoutAudio(s.images, s.transitions, s.transitionDurations, s.timings, s.audios, s.motions, tempDirectory)
+		FFmpeg.MergeTempVideosOldFade(s.images, s.transitionDurations, s.timings, tempDirectory)
+		FFmpeg.AddAudio(s.timings, s.audios, tempDirectory)
+		FFmpeg.CopyFinal(tempDirectory, outputDirectory, final_template_name)
+	}
+
+	fmt.Println("Finished making video...")
+}
+
+func (s slideshow) CreateOverlaidVideo(testVideoDirectory string, finalVideoDirectory string) {
+	FFmpeg.CreateOverlaidVideoForTesting(testVideoDirectory, finalVideoDirectory)
+}
+
+func splitFileNameFromDirectory(slideshowDirectory string) (string, string) {
+	var template_directory_split []string
+
+	template_directory_split = strings.Split(slideshowDirectory, "/")
```

Comment:
The extra code doesn't work for me - it gives me an error. That's why I removed it. I think we need to do more research on why this is the case even though we are both on windows.

Reply:
I updated the code so that it splits for both \\ and / since it should always be one or the other. I don't think there will be any cases where one string has both \\ and /.